prql-compiler/src/tests/test.rs

```diff
@@ -3714,3 +3714,58 @@ fn test_double_stars() {
     "###
     );
 }
+
+#[test]
+fn test_lineage() {
+    // #2627
+    assert_display_snapshot!(compile(r#"
+    from_text """
+    a
+    1
+    2
+    3
+    """
+    derive a = a
```

Comment:
```suggestion
    derive b = a
```

Otherwise the result is a bit confusing since it has duplicate column names

Reply:
Overriding the name was the cause of the error.

And the resulting SQL is even wrong a little, because the first column should be named something else than `a`.

Reply:
Sorry, yes